tools/setup_env.sh

```diff
@@ -0,0 +1,114 @@
+    echo
+}
+
+setup_ip_addresses() {
+    echo "--- Creating a package of IP addresses required for performance tests"
+    sudo python3 ./tools/ips.py --subnet 1.1.1.0/24 --file src/tools/ips.rs --dev_prefix test_zeth
+    if [ $(uname) == "Linux" ]; then
+        sudo python3 ./tools/ips.py --subnet 1.1.1.0/24 --file src/tools/ips.rs --dev lo
+    elif [ $(uname) == "Darwin" ]; then
+        sudo python3 ./tools/ips.py --subnet 1.1.1.0/24 --file src/tools/ips.rs --dev lo0
+    fi
+    echo
+}
+
+setup_initial_node_state() {
+    # Query only after a long delay to account for compilation times and network preparation work
+    ACCOUNT_QUERY_DELAY_MIN=5m
+
+    echo "--- Setting up initial node state, takes at least 5 minutes"
+    echo
+    echo "Spinning up a node instance, please be patient"
+    cargo t setup::testnet::test::run_testnet -- --ignored &
+    echo
+    sleep $ACCOUNT_QUERY_DELAY_MIN
+    echo "--- Querying account info"
+    # Run account query until it responds with "ResponseStatus.SUCCESS" or MAX_ATTEMPTS is reached
+    ACCOUNT_QUERY_TIMEOUT_SEC=5s
+    ACCOUNT_QUERY_MAX_ATTEMPTS=5
+    ACCOUNT_QUERY_NUM_ATTEMPTS=0
+    until [ $ACCOUNT_QUERY_NUM_ATTEMPTS -gt $(($ACCOUNT_QUERY_MAX_ATTEMPTS-1)) ] \
+        || timeout $ACCOUNT_QUERY_TIMEOUT_SEC python3 tools/account_info.py | grep "ResponseStatus.SUCCESS"; do
+        ((ACCOUNT_QUERY_NUM_ATTEMPTS++))
+        echo "Query failed, number of attempts made: $ACCOUNT_QUERY_NUM_ATTEMPTS"
+        echo "Retrying..."
+        sleep $ACCOUNT_QUERY_TIMEOUT_SEC
+    done
+    if [ $ACCOUNT_QUERY_NUM_ATTEMPTS -gt $(($ACCOUNT_QUERY_MAX_ATTEMPTS-1)) ]; then
+        echo "Could not establish a connection with the genesis account. Please try again."
+        exit 1
+    fi
+    echo "Established a connection with the genesis account"
+    echo
+    echo "--- Executing transfer script"
+    python3 tools/transfer.py
+    # Copy the node's files to directory referenced by constant pub const STATEFUL_NODES_DIR
+    cp -a $ZIGGURAT_RIPPLED_TESTNET_DIR $ZIGGURAT_RIPPLED_STATEFUL_DIR
+    echo
+    echo "--- Gracefully stopping the network"
+    kill -2 $(pidof cargo)
+    echo "--- Performing cleanup"
+    # Remove unneeded and temporary files
+    rm $ZIGGURAT_RIPPLED_STATEFUL_DIR/*/rippled.cfg
+    rm -rf $ZIGGURAT_RIPPLED_TESTNET_DIR
+    echo
+}
+
+# Verify the repo location
+if [ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" != "true" ]; then
+    echo "Aborting. Use this script only from the ziggurat/xrpl repo."
+    exit 1
+fi
+REPO_ROOT=`git rev-parse --show-toplevel`
+if [ "`basename $REPO_ROOT`" != "xrpl" ]; then
+    # Wrong root directory, check for rename compared to origin url.
+    ORIGIN_URL=$(git config --local remote.origin.url|sed -n 's#.*/\([^.]*\)\.git#\1#p')
+    if [ "$ORIGIN_URL" != "xrpl" ]; then
+        echo "Aborting. Use this script only from the ziggurat/xrpl repo."
+        exit 1
+    fi
+fi
+
+# Setup the main ziggurat directory in the home directory
```

Comment:
Probably `rm -rf $ZIGGURAT_RIPPLED_DIR` does not mean "Setup the main ziggurat directory" :)